src/ethereum_test_types/block_access_list.py

```diff
@@ -0,0 +1,337 @@
+"""
+Block Access List (BAL) models for EIP-7928.
+
+Following the established pattern in the codebase (AccessList, AuthorizationTuple),
+these are simple data classes that can be composed together.
+"""
+
+from typing import Any, ClassVar, Dict, List, Optional
+
+from pydantic import Field
+
+from ethereum_test_base_types import (
+    Address,
+    Bytes,
+    CamelModel,
+    HexNumber,
+    RLPSerializable,
+    StorageKey,
+)
+from ethereum_test_base_types.serialization import to_serializable_element
+
+
+class BalNonceChange(CamelModel, RLPSerializable):
+    """Represents a nonce change in the block access list."""
+
+    tx_index: int = Field(..., description="Transaction index where the change occurred")
+    post_nonce: int = Field(..., description="Nonce value after the transaction")
+
+    rlp_fields: ClassVar[List[str]] = ["tx_index", "post_nonce"]
+
+
+class BalBalanceChange(CamelModel, RLPSerializable):
+    """Represents a balance change in the block access list."""
+
+    tx_index: int = Field(..., description="Transaction index where the change occurred")
+    post_balance: HexNumber = Field(..., description="Balance after the transaction")
+
+    rlp_fields: ClassVar[List[str]] = ["tx_index", "post_balance"]
+
+
+class BalCodeChange(CamelModel, RLPSerializable):
+    """Represents a code change in the block access list."""
+
+    tx_index: int = Field(..., description="Transaction index where the change occurred")
+    new_code: Bytes = Field(..., description="New code bytes")
+
+    rlp_fields: ClassVar[List[str]] = ["tx_index", "new_code"]
+
+
+class BalStorageChange(CamelModel, RLPSerializable):
+    """Represents a change to a specific storage slot."""
+
+    tx_index: int = Field(..., description="Transaction index where the change occurred")
+    post_value: StorageKey = Field(..., description="Value after the transaction")
+
+    rlp_fields: ClassVar[List[str]] = ["tx_index", "post_value"]
+
+
+class BalStorageSlot(CamelModel, RLPSerializable):
+    """Represents all changes to a specific storage slot."""
+
+    slot: StorageKey = Field(..., description="Storage slot key")
+    slot_changes: List[BalStorageChange] = Field(
+        default_factory=list, description="List of changes to this slot"
+    )
+
+    rlp_fields: ClassVar[List[str]] = ["slot", "slot_changes"]
+
+
+class BalAccountChange(CamelModel, RLPSerializable):
+    """Represents all changes to a specific account in a block."""
+
+    address: Address = Field(..., description="Account address")
+    nonce_changes: Optional[List[BalNonceChange]] = Field(
+        None, description="List of nonce changes"
+    )
```

Comment:
```suggestion
    nonce_changes: List[BalNonceChange] = Field(
        default_factory=list, description="List of nonce changes"
    )
```
Should remove the need for the `to_list` changes.

If you need to know whether a field was set or the default was used we can check `model_fields_set` in the object's instance.